StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedFormViewController.swift

```diff
@@ -0,0 +1,430 @@
+
+    var isUserInteractionEnabled: Bool = true {
+        didSet {
+            if isUserInteractionEnabled != view.isUserInteractionEnabled {
+                sendEventToSubviews(
+                    isUserInteractionEnabled ? .shouldEnableUserInteraction : .shouldDisableUserInteraction,
+                    from: view
+                )
+            }
+            view.isUserInteractionEnabled = isUserInteractionEnabled
+            navigationBar.isUserInteractionEnabled = isUserInteractionEnabled
+        }
+    }
+
+    private func pay(with paymentOption: PaymentOption) {
+        view.endEditing(true)
+        isPaymentInFlight = true
+        error = nil
+        updateError()
+        updatePrimaryButton()
+        isUserInteractionEnabled = false
+
+        // Confirm the payment with the payment option
+        let startTime = NSDate.timeIntervalSinceReferenceDate
+        delegate?.embeddedFormViewControllerShouldConfirm(self, with: paymentOption) { result, deferredIntentConfirmationType in
+            let elapsedTime = NSDate.timeIntervalSinceReferenceDate - startTime
+            DispatchQueue.main.asyncAfter(
+                deadline: .now() + max(PaymentSheetUI.minimumFlightTime - elapsedTime, 0)
+            ) { [self] in
+                analyticsHelper.logPayment(
+                    paymentOption: paymentOption,
+                    result: result,
+                    deferredIntentConfirmationType: deferredIntentConfirmationType
+                )
+
+                self.isPaymentInFlight = false
+                switch result {
+                case .canceled:
+                    // Keep customer on payment sheet
+                    self.updatePrimaryButton()
+                    self.isUserInteractionEnabled = true
+                case .failed(let error):
+#if !canImport(CompositorServices)
+                    UINotificationFeedbackGenerator().notificationOccurred(.error)
+#endif
+                    // Update state
+                    self.isUserInteractionEnabled = true
+                    self.error = error
+                    self.updateError()
+                    self.updatePrimaryButton()
+                    UIAccessibility.post(notification: .layoutChanged, argument: self.errorLabel)
+                case .completed:
+                    // We're done!
+                    let delay: TimeInterval =
+                    self.presentedViewController?.isBeingDismissed == true ? 1 : 0
+                    // Hack: PaymentHandler calls the completion block while SafariVC is still being dismissed - "wait" until it's finished before updating UI
+                    DispatchQueue.main.asyncAfter(deadline: .now() + delay) {
+#if !canImport(CompositorServices)
+                        UINotificationFeedbackGenerator().notificationOccurred(.success)
+#endif
+                        self.primaryButton.update(state: .succeeded, animated: true) {
+                            // Wait a bit before closing the sheet
```

Comment:
lol if you do `git log -S "Wait a bit before closing the sheet"` you'll see this was accidentally copied here and in vertical from the original mc implementation which actually added a delay. This comment doesn't make sense now, delete (here and in vertical too please!)